src/AppBundle/Service/File/Checker/ClamAVChecker.php

```diff
@@ -56,45 +57,37 @@ public function checkFile(UploadableFileInterface $file)
 
-        // wee need some refactor here ?
-        // the call is done here, but the response is not available for PDF scanner
-        // maybe we can remove Types ?
-        // we need to separate logic. Filescanner should be in a different service, injected here
-        // response can be passed to separate file checkers.
-        // With this design I can't tell the PDFChecker to check PDF
-
-
         $file->setScanResult($response);
         $fileName = $file->getUploadedFile()->getClientOriginalName();
-        $fileScannerResult = array_key_exists('file_scanner_result', $response) ?
-             strtoupper(trim($response['file_scanner_result'])) : null;
-        $fileScannerCode = array_key_exists('file_scanner_code', $response) ?
-            strtoupper(trim($response['file_scanner_code'])) : null;
-        $fileScannerMessage = array_key_exists('file_scanner_message', $response) ?
-            strtoupper(trim($response['file_scanner_message']))  : null;
 
+        $fileScannerResult = strtoupper(trim($response['file_scanner_result']));
+        $fileScannerCode = strtoupper(trim($response['file_scanner_code']));
+        $fileScannerMessage = strtoupper(trim($response['file_scanner_message']));
+
+            if ($file instanceof Pdf && $response['file_scanner_result'] !== 'PASS') {
+                $this->logger->warning('file scan result failed in ' . $file->getUploadedFile()->getClientOriginalName() .
+                    ' - ' . $file->getUploadedFile()->getPathName() . '. Scan Result: ' . json_encode($response));
+                throw new RiskyFileException('PDF file scan failed');
+            }
+
+            $this->logger->info('File scan passed for ' . $file->getUploadedFile()->getClientOriginalName() .
+                ' - ' . $file->getUploadedFile()->getPathName() . '. Scan Result: ' . json_encode($response));
         if ($fileScannerResult === 'PASS') {
-            $this->logger->info("Scan result of $fileName: PASS");
+            $this->logger->warning("Scan result of $fileName: PASS");
             return true;
-        } else {
-            throw new RiskyFileException('SCAN FAILED');
         }
 
         $this->logger->warning("Scan result of $fileName: $fileScannerResult, $fileScannerMessage (code $fileScannerCode)");
 
-        if (!is_null($fileScannerCode)) {
-
-            switch ($fileScannerCode) {
-                case 'AV_FAIL':
-                    throw new VirusFoundException('Found virus in file');
-
-                case 'PDF_INVALID_FILE':
-                case 'PDF_BAD_KEYWORD':
-                    throw new RiskyFileException('Invalid PDF');
-            }
-
-            throw new RiskyFileException($fileScannerMessage);
+        switch($fileScannerCode) {
+            case 'AV_FAIL':
+                throw new VirusFoundException('Found virus in file');
+            case 'PDF_INVALID_FILE':
+            case 'PDF_BAD_KEYWORD':
+                throw new RiskyFileException('Invalid PDF');
```

Comment:
do we have a requirement for the message to display ? risky or invalid ? Symfony check for the MIME, so I guess invalid will be rare though